### ISSUE TITLE
Nested AD not defined for conv; any plan?

### ISSUE THREAD

**Issue by tkf** [FluxML/NNlib.jl#119]
Currently, it seems that convolutions do not support higher-order derivatives (which is required for, e.g., [WGAN-GP](https://arxiv.org/abs/1704.00028)):

```julia
julia> using Flux, Tracker

julia> let c = Conv((3, 3), 1 => 1)
           Tracker.gradient(() -> sum(Tracker.jacobian(x -> vec(c(reshape(x, 3, 3, 1, 1))), ones(9)) .^ 2), params(c))
       end
ERROR: Nested AD not defined for conv
```

(The error is from Tracker.jl but it looks like the actual implementation for convolutions and its derivatives are done in NNlib.jl.  So that's why I'm posting here.)

Is there a plan to support this in Flux ecosystem?  Would it happen in NNlib if that's the case?  Also, I guess it's mostly non-interacting with Tracker-to-Zygote transition (as both of them would call primitives defined in NNlib)?

**Comment by MikeInnes** [comment]
cc @tejank10. We worked out gradients for conv transpose at some point, so we can probably just enable this in Tracker.jl

**Comment by tkf** [comment]
Sorry, I haven't found any workaround.

**Comment by mcabbott** [comment]
Fixed by #387:
```julia
julia> using Flux, Zygote

julia> let c = Conv((3, 3), 1 => 1)
           g = Zygote.gradient(() -> sum(Zygote.gradient(x -> sum(c(reshape(x, 3, 3, 1, 1))), ones(Float32, 9))[1] .^ 2), params(c))
           g[c.weight]
       end
3×3×1×1 Array{Float32, 4}:
[:, :, 1, 1] =
  0.492443  -0.690911  -0.0225617
  0.466505   0.951232   0.249715
 -0.286624  -0.248908  -0.0521227

julia> let c = Conv((3, 3), 1 => 1)
           g = Zygote.gradient(() -> sum(Zygote.jacobian(x -> vec(c(reshape(x, 3, 3, 1, 1))), ones(Float32, 9))[1] .^ 2), params(c))
           g[c.weight]
       end
ERROR: Mutating arrays is not supported -- called copyto!(::SubArray{Float32
```
(The most direct translation fails because `Zygote.jacobian` is not differentiable right now.)
